Marlin/Marlin_main.cpp

```diff
@@ -10257,14 +10290,23 @@ void tool_change(const uint8_t tmp_extruder, const float fr_mm_s/*=0.0*/, bool n
       UNUSED(fr_mm_s);
       UNUSED(no_move);
 
-      // Set the new active extruder
-      active_extruder = tmp_extruder;
+      #if ENABLED(SWITCHING_EXTRUDER) && !DONT_SWITCH
+
+        stepper.synchronize();
+        move_extruder_servo(tmp_extruder);
+
+      #elif ENABLED(MK2_MULTIPLEXER)
+
+        if (tmp_extruder >= E_STEPPERS)
+          return invalid_extruder_error(tmp_extruder);
+
+        select_multiplexed_stepper(tmp_extruder);
```

Comment:
I've been looking at this for M701/M702, and it seems to me that the call to `select_multiplexed_stepper` really belongs in the `Planner` class. Several of the methods in the `Planner` class take the extruder index as an argument, which I would think is currently broken since `select_multiplexed_stepper` is not incorporated into those methods.